FlorinPosea/teme/3.5.react-rgb-color-game/src/App.jsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import ColorSquare from "./components/color-square/ColorSquare";
+import "./App.css";
+
+const DEFAULT_SQUARES = 6;
+var winningcolor = "";
+var msg = "";
+
+function App() {
+  const [newList, setNewList] = useState(true);
+  const [colorsList, setColorsList] = useState([]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedColor, setSelectedColor] = useState();
+
+  const createRandomColor = () => {
+    let red = Math.floor(Math.random() * 256);
+    let green = Math.floor(Math.random() * 256);
+    let blue = Math.floor(Math.random() * 256);
+
+    return `rgb(${red}, ${green}, ${blue})`;
+  };
+
+  const getWinningColor = () => {
+    const randomIndex = Math.floor(Math.random() * colorsList.length);
+    return colorsList[randomIndex];
+  };
+
+  const createColorList = (numberOfSquares) => {
+    let colors = [];
+    for (let i = 0; i < numberOfSquares; i++) {
+      const randomColor = createRandomColor();
+      colors.push(randomColor);
+    }
+    return colors;
+  };
+
+  const handleChangeColors = () => {
+    newColors();
+  };
+
+  const newColors = () => {
+    setIsOpen(false);
+    setColorsList(createColorList(DEFAULT_SQUARES));
+    winningcolor = getWinningColor();
+    setSelectedColor();
+  };
+
+  const handleColorPick = (color) => {
+    console.log(color);
+    setIsOpen(true);
+    setSelectedColor(color);
+    onColorPick(color);
+  };
+
+  const onColorPick = (color) => {
+    if (color !== winningcolor) {
+      setColorsList((prevState) => updateColorList(prevState, color));
+      msg = "Try again!";
+    } else {
+      msg = "You won!";
+    }
+  }
+
+
+  const updateColorList = (prevState, color) => {
+    return prevState.map((item) => {
+      if (item === color) {
+        return "rgb(34, 32, 32)";
+      } else {
+        return item;
+      }
+    });
+  } 
+
+
+  if (isOpen === false) {
+    const color = getWinningColor();
+    winningcolor = color;
+  }
```

Comment:
Nu inteleg complet inca idea aici. Tu ai nevoie de un winning color la inceput, on mount. Alte situatii in care winning color se schimba e cand dam click pe buton-ul pentru noi culori. Deci dupa ce schimbi putin cod-ul sa fie cu state (winningColor si msg) cred ca nu o sa mai ai nevoie de bucata asta de cod 
![image](https://user-images.githubusercontent.com/45266485/201001742-e7d9885a-fdbb-4f2d-bb5c-e964d7e79ad9.png)